plaidml/bridge/openvino/ops/deformable_psroi_pooling.cpp

```diff
@@ -0,0 +1,161 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include "plaidml_ops.hpp"
+#include "plaidml_util.hpp"
+
+#include "ngraph/opsets/opset.hpp"
+#include "ngraph/opsets/opset1.hpp"
+
+#include "plaidml/op/op.h"
+
+using namespace plaidml;          // NOLINT[build/namespaces]
+using namespace InferenceEngine;  // NOLINT[build/namespaces]
+
+namespace PlaidMLPlugin {
+
+void registerDeformablePSROIPooling() {
+  registerOp("DeformablePSROIPooling", [](const Context& ctx) {
+    IE_ASSERT(ctx.operands.size() == 3 || ctx.operands.size() == 2);
+    auto has_offset = ctx.operands.size() == 3;
+    auto* layer = ngraph::as_type<ngraph::opset1::DeformablePSROIPooling>(ctx.layer);
+    auto data = ctx.operands.at(0);
+    auto rois = ctx.operands.at(1);
+    auto output_dim = static_cast<int>(layer->get_output_dim());
+    auto group_size = static_cast<int>(layer->get_group_size());
+    auto spatial_scale = layer->get_spatial_scale();
+    auto mode = layer->get_mode();
+    auto spatial_bins_x = static_cast<int>(layer->get_spatial_bins_x());
+    auto spatial_bins_y = static_cast<int>(layer->get_spatial_bins_y());
+    auto trans_std = layer->get_trans_std();
+    auto part_size = layer->get_part_size();
+
+    auto data_shape = data.compute_shape().sizes();
+    auto channels_in = data_shape[1];
+    auto height_in = data_shape[2];
+    auto width_in = data_shape[3];
+    IE_ASSERT(channels_in == output_dim * group_size * group_size);
+
+    auto rois_shape = rois.compute_shape().sizes();
+    auto num_rois = static_cast<int>(rois_shape[0]);
+    auto dim_one = edsl::TensorDim(1);
+
+    rois = edsl::round(rois);
+    auto roi_idx = edsl::index({edsl::TensorDim(2)}, 0) + 1;
+    // Left top ROI corner - (x1, y1)
+    edsl::Tensor roi_lt = edsl::cast(edsl::gather(rois, roi_idx).axis(1), DType::FLOAT32);
```

Comment:
Much as in #1801, I think edsl::gather applied to edsl::index may be less efficient than op::slice. The use here seems fairly small so we could land this anyway, but this is a performance concern to consider in the future. (There is a larger gatherND + index case later which is more substantial, but I think there likely isn't any way to rewrite that using slice given how the index inputs are combined.)

Reply:
Yes, so now, I only replaced `edsl::gather` + continuous `edsl::index` result with `op::slice` and also updated this in `detection_output` and `region_yolo` in #1801. 

For those large `Gather`/`GatherND` cases, I think we can reduce dimensions of gather indices using `scanOp` when it's landed(currently using c++ for loop will significantly increase compilation cost when tensor size is large).